_posts/2017-09-01-railsgirls-fairy-tail.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  Adventures in the Code Ocean – A RailsGirls Fairy Tail
+ref: railsgirls-fairy-tail
+date: 2017-09-01
+lang: en
+orig-lang: en
+author: id-gue
+---
+
+
+Once upon a time there was an [issue](https://github.com/yunity/foodsaving-backend/issues/253) on GitHub. It looks tiny but very useful: just two new fields in the database for saving user feedback. One day two [RailsGirls](https://railsgirlssummerofcode.org/blog/2017-07-13-print_hello_world) discovered the issue. They liked it very much and so they decided that the feedback issue should be their first issue. That was the start of an adventurous journey.
+
+![](https://github.com/yunity/foodsaving-blog/blob/fary-tail-blog-post/images/Issue253.jpg)
+
+
+You must know: the ocean of tech knowledge is incredible large! The two RailsGirls spent years at the shore working at the edge of the programming ocean. Some day they dipped their feet in the water – and it didn’t hurt. <!--more--> On the contrary it was quite exiting! So they started dreaming about going on a ship and sailing the ocean. The opportunity came: a ship of the type Python Django REST with the mission: contribute to foodsaving backend. 
+
+On their first day on board they wanted to start with the feedback issue, but first they had to learn about the harbour. On the third day they thought that they are ready to begin the journey and work on the issue. But to install the sails was much more complicated than they expected. On the fifth day their ship started moving and they had great help by other sailors! But they still didn’t really getting closer to solving the first, tiny issue. Sometimes they lost the track or the wind was very bad. Other days were great – for example when they learned how to navigate with git. 
+
+But even after the two weeks the issue wasn’t solved and the two RailsGirls were getting nervous. Are they too slow? Is the issue too complicated for them – even if it is so tiny? They went on several side tracks to improve their skills and the weeks moved by. 
+
+Then something weird happened: the tiny issue grew! For using the two database fields properly they needed to create permissions and validators – it’s not enough to have a model, but they needed api endpoints and a serialiser and maybe signals? How could they find out if a signal is necessary? And how do all these other things work? It was time to recap and talk with other sailors. Finally they decided to follow a slower but safer course and write tests first. Test driven development is a huge thing on the tech ocean and the two RailsGirls master it slowly but dedicated. 
```

Comment:
"How could they find out if signals _were_ necessary? And how _[-do]_ all these other things _worked_?"
"the two RailsGirls master it slowly but _dedicatedly_."